src/qibolab/instruments/emulator/pulse_simulator.py

```diff
@@ -180,6 +206,32 @@ def play(
             "output_states": output_states,
         }
 
+        if execution_parameters.acquisition_type is AcquisitionType.INTEGRATION:
+            if self.readout_simulator_config is not None:
+                for index, ro_pulse in enumerate(ro_pulse_list):
+                    # obtain qubit with its qubitID
+                    qubit = qubits[get_qubit(ro_pulse.qubit, qubits)]
+                    # bare_resonator_frequency is prepared considering readout pulse frequency = lambshifted readout pulse frequency,
+                    # such that the V_I/V_Q are maximally seperated (i.e.: bare_resonator_frequency = lambshifted readout pulse frequency + lamb shift)
+                    # solve quadratic equation for bare_resonator frequency
+                    readout = ReadoutSimulator(
+                        qubit=qubit, **self.readout_simulator_config
+                    )
+                    values = np.array(samples[ro_pulse.qubit])
+                    for i in range(len(values)):
+                        values = values.astype(np.complex128)
```

Comment:
No need to recast every iteration
```suggestion
                    values = np.array(samples[ro_pulse.qubit]).astype(np.complex128)
                    for i in range(len(values)):
```